### PR TITLE
Support new Purpose header field in TCF Banner

### PR DESCRIPTION
**Pull request by gilluminate**
Closes [PROD-2699](https://ethyca.atlassian.net/browse/PROD-2699)

### Description Of Changes

* Add new field in Admin UI > Experiences > TCF Experience > Translations form to support updating the Purpose header on the TCF banner
* Update fides-tcf.js to consume the translated string
* Update Cypress tests to include the new header

<hr/>

CleanShot 2024-08-29 at 15 41 17@2x

<hr/>

CleanShot 2024-08-29 at 15 41 41@2x

<hr/>

### Code Changes

* Lots of updates to support this change in Typing, helpers, tests, etc.
* Additional field in the translations form (TCF only)
* include translation in FidesJS component

### Steps to Confirm

1. In Admin UI, visit a TCF experience and click the English language to open the translation form
2. Update or add text to the "Purpose header (optional)" field
3. Save the translation and then Save the experience
4. Visit the demo page for the TCF experience (eg. `http://localhost:3001/fides-js-demo.html?geolocation=eea`)
5. Ensure that the text typed in appears above the Purpose list in the right column of the TCF banner.

### Pre-Merge Checklist

* [x] All CI Pipelines Succeeded
* [x] Issue Requirements are Met
* [ ] Backend changes are ready for merging

NOTE: since this will be hotfixed in the current release, no separate entry in the CHANGELOG is needed.

[PROD-2699]: https://ethyca.atlassian.net/browse/PROD-2699?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: clients/fides-js/src/lib/i18n/i18n-utils.ts [hunk 1]
```diff
@@ -117,6 +118,7 @@ function extractMessagesFromExperienceConfig(
     const locale = experienceConfig.language || DEFAULT_LOCALE;
     const messages: Messages = {};
     EXPERIENCE_TRANSLATION_FIELDS.forEach((key) => {
+      // @ts-expect-error EXPERIENCE_TRANSLATION_FIELDS is a const array
```
Comment (gilluminate) [inline clients/fides-js/src/lib/i18n/i18n-utils.ts:121]: unrelated fix, just to get rid of the error message in VS Code.

File: clients/admin-ui/src/features/privacy-experience/form/helpers.tsx [hunk 2]
```diff
@@ -162,7 +163,24 @@ export const getTranslationFormFields = (
       modal_link_label: { included: true },
     };
   }
-  // For TCF overlay / default
+
+  if (component === ComponentType.TCF_OVERLAY) {
```
[the remark this answers is not in the document]
Comment (eastandwestwind) [reply clients/admin-ui/src/features/privacy-experience/form/helpers.tsx:167]: makes sense 👍

**Review by eastandwestwind — APPROVED** [review]

**Comment by pattisdr** [comment]
Tested locally alongside https://github.com/ethyca/fidesplus/pull/1589, this is looking good. 

- Both __VENDOR_COUNT__ paragraphs are rendering properly
- Vendor count buttons are working on first load and with other translations, and open the modal
- Everything is nice and scrollable 
- No issues if there is a description that does not have a __VENDOR_COUNT__ placeholder, as could be the case with customers that have already edited their banners
- purpose headers rendering fine and present in the admin UI
